### PR TITLE
Start of loop in a linked list

### LINKED ISSUE #1606: Find Start of a Loop in a Linked List JAVA
This will be done using hare and tortoise approach.

### PR DESCRIPTION
**Pull request by ishaan2908**
Fixes #1606 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added a method to find starting node of a loop in a linked list

#### Languages Used:
- JAVA

#### Files Added:
- StartOfLoop.java

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Linked_List/StartOfLoop.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import java.util.Scanner;
```
Comment (MastersAbh) [inline Linked_List/StartOfLoop.java]: Please add problem statement on top of the code.

File: Linked_List/StartOfLoop.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	{
+		int data;
+		Node next;
+		Node(int d)
+		{
+			data = d;
+			next = null;
+		}
+	}
+
```
Comment (MastersAbh) [inline Linked_List/StartOfLoop.java]: Give 4 space indentation across the file

File: Linked_List/StartOfLoop.java (hunk 1 above)
Comment (ishaan2908) [reply Linked_List/StartOfLoop.java]: I have made the required changes.

File: Linked_List/StartOfLoop.java (hunk 2 above)
Comment (ishaan2908) [reply Linked_List/StartOfLoop.java]: I have made the required changes.

**Comment by ishaan2908** [comment]
I have made the requested changes. Please review! @MastersAbh

File: Linked_List/StartOfLoop.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    static void createLoop(int p)
+    {
+        // For demonstration, we will link the last node with the element at position p.
+        Node current = head, tail = head;
+        while(tail.next != null)
+            tail = tail.next;
+        
+        int count = 1;
+        while(current.next != null && count < p)
+		{
```
Comment (MastersAbh) [inline Linked_List/StartOfLoop.java]: Please fix indentation for while loop. The { should start from below the while statement

**Comment by ishaan2908** [comment]
Done with the changes! @MastersAbh

**Pull request merged**